src/Diagnostics.DataProviders/TokenService/TokenServiceBase.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System.Threading.Tasks;
+using System;
+using System.Linq;
+using Diagnostics.Logger;
+
+namespace Diagnostics.DataProviders.TokenService
+{
+    public class TokenServiceBase
```

Comment:
I suggest making the class an `abstract` class and making all the properties abstract too to force any inheritors to provide values

Reply:
I am not sure if I understood this fully, so correct me if I'm wrong. 

>  making all the properties abstract too to force any inheritors to provide values

Which properties are you saying should be abstract? The idea is that derived classes should provide values that are needed for AAD such as `ClientCredential` , correct? 

I made the `AuthorizationToken` to have private set so that only the `StartTokenRefresh` can set the value and derived classes cannot modify this.